utils/k8s.go

```diff
@@ -0,0 +1,221 @@
+// +build k8s
+
+package utils
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/containernetworking/cni/pkg/ipam"
+	"github.com/containernetworking/cni/pkg/skel"
+	"github.com/containernetworking/cni/pkg/types"
+	"github.com/projectcalico/libcalico/lib"
+)
+
+//
+//import (
+//	"fmt"
+//	"os"
+//
+//	k8sRestClient "k8s.io/kubernetes/pkg/client/restclient"
+//	k8sClient "k8s.io/kubernetes/pkg/client/unversioned"
+//)
+//
+
+func RunningUnderK8s(args *skel.CmdArgs) (bool, error) {
+
+	// Determine if running under k8s by checking the CNI args
+	k8sArgs := K8sArgs{}
+	if args.Args != "" {
+		err := LoadArgs(args.Args, &k8sArgs)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return string(k8sArgs.K8S_POD_NAMESPACE) != "" && string(k8sArgs.K8S_POD_NAME) != "", nil
+}
+
+func CmdAddK8s(args *skel.CmdArgs, k8sArgs K8sArgs, conf NetConf) (*types.Result, error) {
```

Comment:
I'd like to think and then chat a bit more about how we're structuring the different CmdAdd calls for different orchestrators.

Mostly:
- Ensuring we're not duplicating excessive amounts of code as we go on.
- Ensuring that the differences between orchestrators don't get lost in the abundance of common code between them.
- As an example / case study, how easy will it be to fit Mesos in to this model?